src/licensedcode/models.py

```diff
@@ -1866,6 +1897,12 @@ def validate(self, licensing=None, thorough=False):
             if not check_is_list_of_strings(self.referenced_filenames):
                 yield 'referenced_filenames must be a list of strings'
 
+            if (self.is_license_clue or self.is_license_intro) and self.is_required_phrase:
```

Comment:
This is to prevent marking of license clues or license intro rules as required phrases as by definition these two types of rules are either non-specific intros or weak clues as rules, which cannot be considered required phrases for any specific license expressions.